server/modlog.ts

```diff
@@ -153,21 +155,32 @@ export class Modlog {
 	readonly globalPunishmentsSearchQuery: Database.Statement<[string, string, string, number, ...string[]]>;
 	readonly insertionTransaction: Database.Transaction;
 
+	readonly useFTS: boolean;
+
 	constructor(flatFilePath: string, databasePath: string) {
 		this.logPath = flatFilePath;
+		this.useFTS = Config.modlogfts;
 
 		this.database = new Database(databasePath);
 		this.database.exec("PRAGMA foreign_keys = ON;");
-		this.database.exec(FS(MODLOG_SCHEMA_PATH).readIfExistsSync()); // Set up tables, etc.
+
+		// Set up tables, etc
+		this.database.exec(
+			FS(this.useFTS ? MODLOG_SCHEMA_WITH_FTS_PATH : MODLOG_SCHEMA_PATH).readIfExistsSync()
+		);
 		this.database.function('regex', {deterministic: true}, (regexString, toMatch) => {
 			return Number(RegExp(regexString, 'i').test(toMatch));
 		});
 
 
-		this.modlogInsertionQuery = this.database.prepare(
-			`INSERT INTO modlog (timestamp, roomid, visual_roomid, action, userid, autoconfirmed_userid, ip, action_taker_userid, note)` +
-			` VALUES ($time, $roomID, $visualRoomID, $action, $userid, $autoconfirmedID, $ip, $loggedBy, $note)`
-		);
+		let insertionQuerySource = `INSERT INTO modlog (timestamp, roomid, visual_roomid, action, userid, autoconfirmed_userid, ip, action_taker_userid, note)`;
+		insertionQuerySource += ` VALUES ($time, $roomID, $visualRoomID, $action, $userid, $autoconfirmedID, $ip, $loggedBy, $note)`;
+		if (this.useFTS) {
+			insertionQuerySource += `INSERT INTO modlog_fts (timestamp, roomid, visual_roomid, action, userid, autoconfirmed_userid, ip, action_taker_userid, note)`;
```

Comment:
I haven't tried this yet but this doesn't look right to me. `modlog_fts` should have columns `note` and (implicitly) `rowid`. When you insert a modlog entry, get its `modlog_id` (from SQL, `last_insert_rowid()`) and `INSERT INTO modlog_fts (rowid, note) VALUES (<last row id>, <same note as you inserted into the modlog table>)`. In `modlog_fts`, `note` is the tokenized version whereas in `modlog` it is the literal note string you specified